### PR TITLE
LFS: part 10, sysdeps

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

File: options/posix/generic/sys-socket-stubs.cpp [hunk 1]
```diff
@@ -98,6 +90,7 @@ ssize_t recv(int sockfd, void *__restrict buf, size_t len, int flags) {
 
 ssize_t recvfrom(int sockfd, void *__restrict buf, size_t len, int flags,
 		struct sockaddr *__restrict src_addr, socklen_t *__restrict addrlen) {
+#ifndef __linux__
```
Comment (ikbenlike) [inline options/posix/generic/sys-socket-stubs.cpp:93]: should this not take place in the implementation of the sysdep?

File: options/posix/generic/sys-socket-stubs.cpp [hunk 2]
```diff
@@ -140,6 +142,7 @@ ssize_t send(int fd, const void *buffer, size_t size, int flags) {
 
 ssize_t sendto(int fd, const void *buffer, size_t size, int flags,
 		const struct sockaddr *sock_addr, socklen_t addr_length) {
+#ifndef __linux__
```
Comment (ikbenlike) [inline options/posix/generic/sys-socket-stubs.cpp:145]: same as with `recvfrom`

**Review by ikbenlike — COMMENTED** [review]
from `recvfrom(2)`:
> The recv() call is normally used only on a connected socket (see connect(2) and is identical to recvfrom() with a NULL src_addr argument.

I am unsure if mlibc already provides `recv`, however it might be worth it to simplify the implementation of it like this (and if not yet provided, it would be relatively simple to do so)

**Comment by ikbenlike** [comment]
I just checked and `recv` is already implemented as a call to `recvfrom`, and the fallback in case there is no sysdep is now handled more elegantly. personally I can't think of anything else that should be changed in this PR

**Review by Dennisbonke — APPROVED** [review]
LGTM

File: options/posix/include/mlibc/posix-sysdeps.hpp [hunk 3]
```diff
@@ -166,7 +168,7 @@ int sys_vm_unmap(void *pointer, size_t size);
 // as an infinite timeout, making sigtimedwait(..., nullptr) equivalent to sigwaitinfo(...)
 [[gnu::weak]] int sys_sigtimedwait(const sigset_t *__restrict set, siginfo_t *__restrict info, const struct timespec *__restrict timeout, int *out_signal);
 [[gnu::weak]] int sys_kill(int, int);
-[[gnu::weak]] int sys_accept(int fd, int *newfd, struct sockaddr *addr_ptr, socklen_t *addr_length);
+[[gnu::weak]] int sys_accept(int fd, int *newfd, struct sockaddr *addr_ptr, socklen_t *addr_length, int flags);
```
Comment (Geertiebear) [inline options/posix/include/mlibc/posix-sysdeps.hpp:171]: The change of this sysdep should be propagated to other ports as well. Also make sure that the functionality stays the same (so copy the old ``fcntl()`` trick to the sysdeps as well).

**Review by Geertiebear — CHANGES_REQUESTED** [review]

**Review by Geertiebear — APPROVED** [review]
LGTM now.

**Pull request merged**